### PR TITLE
3 fixes (SQL Server Statement, pay attention to conversation.maxRetrievable, parameter issue in sql)

### PR DESCRIPTION
**Pull request by mightymop**
SQL Server does not know the keyword "limit" 
> change query to use "top" function so solve SqlExceptions

### CODE REVIEW

File: src/java/com/reucon/openfire/plugin/archive/impl/JdbcPersistenceManager.java [hunk 1]
```diff
@@ -489,22 +489,31 @@ private int bindConversationParameters(Date startDate, Date endDate, String owne
             if (xmppResultSet.getIndex() != null) {
                 firstIndex = xmppResultSet.getIndex();
             } else if (xmppResultSet.getAfter() != null) {
-                final Long needle;
+                Long needle;
                 if ( useStableID ) {
                     needle = ConversationManager.getMessageIdForStableId( new JID(ownerJid), xmppResultSet.getAfter() );
+                    if (needle==null)
+                    	needle = new Long((new java.util.Date()).getTime());
```
Comment (guusdk) [inline src/java/com/reucon/openfire/plugin/archive/impl/JdbcPersistenceManager.java]: Can you please use a consistent amount of indentation, and always use brackets (even if there's just one line in the if-statement)

**Comment by guusdk** [comment]
Thanks! This PR appears to fix three different issues. Can you please create separate issues in  https://github.com/igniterealtime/openfire-monitoring-plugin/issues? Next, please rewrite your commit message, so that each commit references the issue that it fixes. Lastly, please add the fixed issues to the changelog that is part of the source code for this plugin!

**Comment by mightymop** [comment]
Ok, i am going to close this and will open 3 new PRs

**Comment by mightymop** [comment]
https://github.com/igniterealtime/openfire-monitoring-plugin/pull/65
https://github.com/igniterealtime/openfire-monitoring-plugin/pull/66
https://github.com/igniterealtime/openfire-monitoring-plugin/pull/68

**Pull request closed without merging**